src/main/groovy/de/iteratec/osm/measurement/environment/wptserver/WptResultXml.groovy

```diff
@@ -127,6 +127,45 @@ class WptResultXml {
         }
     }
 
+    int getResultCodeForStep(int stepZeroBasedIndex) {
```

Comment:
You can use `getStepNode` from above here and in the next methods to avoid duplicating the version-specific logic.

Reply:
I think that the usage of getStepNode doesn't help to avoid the version-specific logic. The path of the properties in the XML is still depending on the version, even if i use the getStepNode method.

Reply:
Done. :)